src/Npgsql/NpgsqlSchema.cs

```diff
@@ -16,33 +16,38 @@ namespace Npgsql
     static class NpgsqlSchema
     {
         public static DataTable GetSchema(NpgsqlConnection conn, string? collectionName, string?[]? restrictions)
+            => GetSchemaAsync(conn, collectionName, restrictions, false).GetAwaiter().GetResult();
+
+        public static Task<DataTable> GetSchemaAsync(NpgsqlConnection conn, string? collectionName, string?[]? restrictions, bool async = true, CancellationToken cancellationToken = default)
         {
             if (collectionName is null)
                 throw new ArgumentNullException(nameof(collectionName));
             if (collectionName.Length == 0)
                 throw new ArgumentException("Collection name cannot be empty.", nameof(collectionName));
 
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
This already happens at the public API surface, so I don't think it needs to be here again.